pkg/cloudprovider/aws/object_store.go

```diff
@@ -85,6 +85,10 @@ func (o *objectStore) Init(config map[string]string) error {
 		WithS3ForcePathStyle(s3ForcePathStyle)
 
 	if s3URL != "" {
+		if !IsValidS3URLScheme(s3URL) {
```

Comment:
I think we could make this a little bit more robust by using `net/url`'s `Parse` function, which will return an error if it's not a valid URL, and then we can check the returned struct's `Scheme` property to explicitly ensure it's http/https.  WDYT?

Reply:
I agree! I'll push a change shortly.